troupe.go

```diff
@@ -58,73 +61,106 @@ func NewTroupe(cfg Config) (*Troupe, error) {
 		if b, err = NewActor(bCfg); err != nil {
 			return nil, err
 		}
+		b.ID = len(Actors)
 		Actors = append(Actors, b)
 	}
 	return &Troupe{
 		Actors:             Actors,
 		minActors:          cfg.Min,
 		maxActors:          cfg.Max,
 		defaultActorConfig: bCfg,
+		r:                  rand.New(rand.NewSource(time.Now().UnixNano())),
```

Comment:
I hadn't really considered that - I will say i'm still not sure if i'll stick with the random approach or not. I'm still trying to get a good idea on if the speed of assigning work via random is tenable compared to the efficiency of getting work that I'd get with the current priority approach (IE: I'd rather trade off a little bit of time and efficiency getting work to an Actor, if I can ensure that work goes to the actor who can most quickly get the work done).